dashboard/app/helpers/application_helper.rb

```diff
@@ -63,14 +63,11 @@ def activity_css_class(result)
   end
 
   def level_info(user, script_level, user_levels)
-    result = nil
     if user
-      ul = user_levels[script_level.level_id]
-      result = ul.try(:best_result) if ul
+      user_levels[script_level.level_id].try(:best_result) || 0
     else
-      result = client_state.level_progress(script_level)
+      0
```

Comment:
Agreed, I tried to avoid changing signatures as much as possible.  In this case it's only used in the one place (verified with full text search for usages).  Good call on moving it to UsersHelper, I will update.